lms/static/js/ccx/schedule.js

```diff
@@ -343,13 +376,22 @@ var edx = edx || {};
 	      children.hide();
 	    }
 	    else {
-				$(this).attr('aria-expanded', 'true');
-	      $(this).find(".fa-caret-right").removeClass('fa-caret-right').addClass('fa-caret-down');
-	      row.removeClass('collapsed').addClass('expanded');
-	      children.filter('.collapsed').each(function() {
-		children = children.not(self.get_children(this));
-	      });
-	      children.show();
+                $(this).attr('aria-expanded', 'true');
+                $(this).find(".fa-caret-right").removeClass('fa-caret-right').addClass('fa-caret-down');
+                row.removeClass('collapsed').addClass('expanded');
+                var depth = $(row).data('depth');
+                var $childNodes = children.filter('.collapsed');
+                if ($childNodes.length <= 0) {
+                  children.show();
+                } else {
+									// this will expand units.
```

Comment:
On further inspection, it looks like this *whole* file freely mixes spaces and tabs. After this pull request is merged, please make a follow-on pull request that does nothing but change all tabs to spaces in this file, and correct indentation properly. (Having a pull request that changes *only* whitespace, and none of the actual code in the file, makes it *much* easier to review.)